src/main/java/com/stirante/runechanger/sourcestore/impl/LolalyticsSource.java

```diff
@@ -0,0 +1,200 @@
+package com.stirante.runechanger.sourcestore.impl;
+
+import com.google.gson.Gson;
+import com.stirante.runechanger.model.client.*;
+import com.stirante.runechanger.sourcestore.RuneSource;
+import com.stirante.runechanger.util.FxUtils;
+import javafx.collections.ObservableList;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class LolalyticsSource implements RuneSource {
+    private static final Logger log = LoggerFactory.getLogger(ChampionGGSource.class);
+    private final static String CHAMPION_URL = "https://api.op.lol/champion/3/?patch=%PATCH%tier=platinum_plus&queue=420&region=all&cid=%CHAMPIONID%&lane=%LANE%";
+    private static final int TIMEOUT = 10000;
+    private Map<Rune, int[]> runeMap;
+
+    private void downloadRunes(Champion champion, ObservableList<RunePage> pages) {
+        final String[] lanes = {"Top", "Jungle", "Middle", "Bottom", "Support"};
+        try {
+            for (String lane : lanes) {
+                final URL url = new URL(CHAMPION_URL
+                        .replace("%PATCH", Patch.getLatest(1).get(0).toString())
+                        .replace("%CHAMPIONID%", Integer.toString(champion.getId()))
+                        .replace("%LANE%", lane.toLowerCase()));
+                HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+                conn.connect();
+                LolalyticsResult jsonData = new Gson().fromJson(new InputStreamReader(conn.getInputStream()), LolalyticsResult.class);
+                conn.getInputStream().close();
+                ConvertedDataPair convertedDataPair = convertRunes(jsonData.display);
+                RunePage runePage = calculateRunes(convertedDataPair, 0);
+                if(runePage != null) {
+                    runePage.setChampion(champion);
+                    runePage.setName(lane);
+                    runePage.setSourceName(this.getSourceName());
+                    runePage.setSource("https://lolalytics.com/");
```

Comment:
The source should be something, that identifies the literal source of the rune page. In this case, it would be URL but with `/lol/missfortune/?lane=bottom`